examples/image_segmentation/ios/ImageSegmenter/ViewContoller/MediaLibraryViewController.swift

```diff
@@ -0,0 +1,384 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AVKit
+import MediaPipeTasksVision
+import UIKit
+import Metal
+import MetalKit
+import MetalPerformanceShaders
+
+/**
+ * The view controller is responsible for performing segmention on videos or images selected by the user from the device media library and
+ * presenting them with the new backgrourd of the image to the user.
+ */
+class MediaLibraryViewController: UIViewController {
+
+  // MARK: Constants
+  private struct Constants {
+    static let edgeOffset: CGFloat = 2.0
+    static let inferenceTimeIntervalInMilliseconds = 200.0
+    static let milliSeconds = 1000.0
+    static let savedPhotosNotAvailableText = "Saved photos album is not available."
+    static let mediaEmptyText =
+    "Click + to add an image or a video to begin running the image sengmentation."
+    static let pickFromGalleryButtonInset: CGFloat = 10.0
+  }
+  // MARK: Face Segmenter Service
+  weak var inferenceResultDeliveryDelegate: InferenceResultDeliveryDelegate?
+
+  // MARK: Controllers that manage functionality
+  private lazy var pickerController = UIImagePickerController()
+  private var playerViewController: AVPlayerViewController?
+
+  private let backgroundQueue = DispatchQueue(label: "com.google.mediapipe.cameraController.backgroundQueue")
+  private let imageSegmenterServiceQueue = DispatchQueue(
+    label: "com.google.mediapipe.cameraController.imageSegmenterServiceQueue",
+    attributes: .concurrent)
+
+  // MARK: Face Segmenter Service
+  private var imageSegmenterService: ImageSegmenterService?
+//  private var imageSegmenterService: ImageSegmenterService? {
```

Comment:
Is there a reason for commenting this code? We are modifying the image segmenter service from different threads when we switch between modes. This is necessary to make sure two threads don't update it simultaneously.